### PR TITLE
Add support for NumPy

### PR DESCRIPTION
**Pull request by andreittr**
This change set adds support for the numpy extension in the main Python build.

Part of work on Python 3.10
- https://github.com/unikraft/lib-python3/pull/15

and thus contains all its commits as well.
Only 5ad2760d1de0ccd695a579bad057fe981f98fb3f is the actual numpy support. (Will rebase once Python PR is merged)

The NumPy microlibrary is a separate PR:
- https://github.com/unikraft/lib-python-numpy/pull/1

### CODE REVIEW

File: Config.uk [hunk 1]
```diff
@@ -80,6 +80,12 @@ config LIBPYTHON3_EXTENSION_ZLIB
 	bool "zlib"
 	default y
 endif
+
+if LIBPYTHON_NUMPY
+config LIBPYTHON3_EXTENSION_NUMPY
```
Comment (StefanJum) [inline Config.uk:86]: Why not do a `select LIBPYTHON_NUMPY` if `LIBPYTHON3_EXTENSION_NUMPY` is selected?

**Review by StefanJum — CHANGES_REQUESTED** [review]

File: Config.uk (hunk 1 above)
Comment (andreittr) [reply Config.uk:86]: This follows the convention used for all other extensions, and I don't see numpy as special enough to break the pattern.
In addition, having the numpy extension option show up irrespective of whether numpy is present in the build or not is definitely misleading, and if the default remains 'y' (as is the case for all other extensions) the build will fail at link time.

**Review by StefanJum — APPROVED** [review]
All good @andreittr, will add the tag after #15 is merged.

**Review by mariasfiraiala — APPROVED** [review]
Works well, thanks!

Reviewed-by: Maria Sfiraiala <maria.sfiraiala@gmail.com>

**Comment by StefanJum** [comment]
@andreittr you can rebase this now, #15 is merged :tada:

**Review by StefanJum — APPROVED** [review]
All good, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**